### PR TITLE
feat: rename MinGenInfo to be compatible with birth rounds.

### LINKED ISSUE #11720: Rename MinGenInfo to be compatible with birth rounds
We can reuse the `MinGenInfo` object when we have birth rounds, no serialization changes required. In order to do so without going crazy, we need to rename things so that it makes sense either with generations or birth rounds.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #11720

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MinimumJudgeInfo.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Records the minimum ancient threshold for all judges in a particular round.
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MinimumJudgeInfo.java:27]: Wouldn't it be the minimum ancient `indicator`?   The minimum indicator of  judges in round `latestConsensusRound-26+1` becomes the ancient threshold.    This is storing the minimum indicator for the given round, not the minimum threshold for round -26+1

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusRounds.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public class ConsensusRounds {
     /** consensus configuration */
     private final ConsensusConfig config;
     /** stores the minimum generation for all decided and non-expired rounds */
-    private final SequentialRingBuffer<MinGenInfo> minGenStorage;
+    private final SequentialRingBuffer<MinimumJudgeInfo> minGenStorage;
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusRounds.java:40]: Optional: this member variable name, and many comments/parameter names in this class could also be updated.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java [hunk 3]
```diff
@@ -108,10 +102,10 @@ public long round() {
     }
 
     /**
-     * @return the round generation numbers for all non-ancient rounds
+     * @return for each non-ancient round, the minimum ancient threshold of the round's judges
      */
-    public @NonNull List<MinGenInfo> minGens() {
-        return minGens;
+    public @NonNull List<MinimumJudgeInfo> minGens() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java:107]: method name could be updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java [hunk 4]
```diff
@@ -131,57 +125,61 @@ public long nextConsensusNumber() {
     /**
      * Returns the minimum generation below which all events are ancient
      *
-     * @param roundsNonAncient
-     * 		the number of non-ancient rounds
+     * @param roundsNonAncient the number of non-ancient rounds
      * @return minimum non-ancient generation
      */
     public long getMinimumGenerationNonAncient(final int roundsNonAncient) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java:131]: method name could be updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java [hunk 5]
```diff
@@ -131,57 +125,61 @@ public long nextConsensusNumber() {
...
+                roundsNonAncient, round, this::getMinimumJudgeAncientThreshold);
     }
 
     /**
-     * The minimum generation of famous witnesses for the round specified. This method only looks at non-ancient rounds
-     * contained within this state.
+     * The minimum ancient threshold of famous witnesses (i.e. judges) for the round specified. This method only looks
+     * at non-ancient rounds contained within this state.
      *
      * @param round the round whose minimum generation will be returned
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java:140]: the param doc still mentions generation

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/RoundElections.java [hunk 6]
```diff
@@ -118,10 +118,10 @@ public long getMinGeneration() {
     }
 
     /**
-     * @return create a {@link MinGenInfo} instance for this round
+     * @return create a {@link MinimumJudgeInfo} instance for this round
      */
-    public @NonNull MinGenInfo creatMinGenInfo() {
-        return new MinGenInfo(round, getMinGeneration());
+    public @NonNull MinimumJudgeInfo creatMinGenInfo() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/RoundElections.java:123]: method name could be updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MinimumJudgeInfo.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+ */
+public record MinimumJudgeInfo(long round, long minimumJudgeAncientThreshold) {
+
+    /**
+     * The maximum permissible size of a list of {@link MinimumJudgeInfo} lists as deserialized by
+     * {@link #deserializeList(SerializableDataInputStream)}. Is an upper bound for
+     * {@link com.swirlds.platform.consensus.ConsensusConfig#roundsNonAncient()}, choices for this config that exceed
+     * this value will result in an exception being thrown when deserializeList is called.
+     */
+    public static int MAX_MINIMUM_JUDGE_INFO_SIZE = 32;
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MinimumJudgeInfo.java:42]: may be `final`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java [hunk 8]
```diff
@@ -90,7 +90,7 @@ private static final class ClassVersion {
     /**
      * the minimum generation of famous witnesses per round
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java:91]: doc can be updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java [hunk 9]
```diff
@@ -340,11 +340,11 @@ public PlatformData setConsensusTimestamp(final Instant consensusTimestamp) {
      *
      * @return minimum generation info list
      */
-    public List<MinGenInfo> getMinGenInfo() {
+    public List<MinimumJudgeInfo> getMinGenInfo() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java:343]: method name and doc can be updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java [hunk 10]
```diff
@@ -502,7 +502,7 @@ public String toString() {
                 .append("hashEventsCons", hashEventsCons)
                 .append("events", events)
                 .append("consensusTimestamp", consensusTimestamp)
-                .append("minGenInfo", minGenInfo)
+                .append("minGenInfo", minimumJudgeInfo)
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java:505]: fieldName should be updated to match new member variable name

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java [hunk 11]
```diff
@@ -339,7 +339,7 @@ public void setConsensusTimestamp(@NonNull final Instant consensusTimestamp) {
      * @return minimum generation info list, or null if this is a genesis state
      */
     @Nullable
-    public List<MinGenInfo> getMinGenInfo() {
+    public List<MinimumJudgeInfo> getMinGenInfo() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java:342]: method name and doc could be updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java [hunk 12]
```diff
@@ -272,7 +272,7 @@ public String getInfoString(final int hashDepth) {
                 .addRow("Creation version:", platformState.getCreationSoftwareVersion())
                 .addRow("Epoch mnemonic:", epochHash == null ? "null" : epochHash.toMnemonic())
                 .addRow("Epoch hash:", epochHash)
-                .addRow("Min gen hash code:", minGenInfo == null ? "null" : minGenInfo.hashCode())
+                .addRow("Min gen hash code:", minimumJudgeInfo == null ? "null" : minimumJudgeInfo.hashCode())
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java:275]: row label should be changed

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/PlatformDataTests.java [hunk 13]
```diff
@@ -58,10 +58,10 @@ static void beforeAll() throws ConstructableRegistryException {
     private static PlatformData generateRandomPlatformData(final Random random) {
         final int randomBound = 10_000;
 
-        final List<MinGenInfo> minGenInfo = new LinkedList<>();
-        final int minGenInfoSize = random.nextInt(1, MinGenInfo.MAX_MINGEN_INFO_SIZE);
+        final List<MinimumJudgeInfo> minimumJudgeInfo = new LinkedList<>();
+        final int minGenInfoSize = random.nextInt(1, MinimumJudgeInfo.MAX_MINIMUM_JUDGE_INFO_SIZE);
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/PlatformDataTests.java:62]: `minGenInfoSize` could be renamed

**Review by alittley — APPROVED** [review]

**Review by lpetrovic05 — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MinimumJudgeInfo.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MinimumJudgeInfo.java:27]: changed to `indicator`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusRounds.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusRounds.java:40]: I renamed the variables and methods, and I fixed the javadocs. I specifically did not alter the business logic that does stuff like compare `event.getGeneration()`, as that is outside the scope of this PR.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java:131]: I don't think it's right to rename this method until the business logic inside it changes as well. I'm intentionally avoiding all changes to business logic in this PR.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java:140]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java (hunk 8 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java:91]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java (hunk 9 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java:343]: updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java (hunk 10 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java:505]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java (hunk 11 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java:342]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java (hunk 12 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java:275]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/RoundElections.java [hunk 14]
```diff
@@ -120,7 +120,7 @@ public long getMinGeneration() {
     /**
      * @return create a {@link MinimumJudgeInfo} instance for this round
      */
-    public @NonNull MinimumJudgeInfo creatMinGenInfo() {
+    public @NonNull MinimumJudgeInfo creatMinimumJudgeInfo() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/RoundElections.java:123]: Whoops, "creat" -> "create"

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/RoundElections.java]: fixed

**Review by alittley — APPROVED** [review]

**Pull request merged**